oio/xcute/orchestrator.py

```diff
@@ -170,15 +172,16 @@ def handle_running_job(self, job_info):
             args=(job_id, tasks_counter))
         total_tasks_thread.start()
 
-    def dispatch_job(self, job_id, job_type, job_config, job_tasks,
-                     beanstalkd_workers):
+    def dispatch_job(self, job_id, job_type, job_config, job_tasks):
         """
             Dispatch all of a job's tasks
         """
 
         self.logger.info('Start dispatching job (job_id=%s)', job_id)
 
         try:
+            beanstalkd_workers = self.get_beanstalkd_workers()
```

Comment:
I forgot to mention it, but the iterator could be shared by all the jobs. Any pros/cons for one iterator per job? If we had one for all the jobs, the loadbalancing would probably be a bit more efficient

Reply:
I like this idea and I had already thought about it. But the way I implemented it, I couldn't make it work with 2 jobs using this same iterator.
```
ValueError: generator already executing
```